python/src/uagents/experimental/dialogues/__init__.py

```diff
@@ -146,14 +145,13 @@ async def handler(ctx: Context, sender: str, message: MessageModel):
     def __init__(
         self,
         name: str,
-        agent_address: str,  # tbd: storage naming and handling
-        version: Optional[str] = None,
-        nodes: List[Node] | None = None,
-        edges: List[Edge] | None = None,
+        storage: StorageAPI = None,
```

Comment:
Also could we hide it away by auto-setting it on agent.include() (assuming agent.storage will always exist?)

Reply:
Unfortunately this can't be auto set (at least I don't see how at the current moment) as the `Protocol` / `Dialogue` will only ever come in contact with the agent during `agent.include(<Dialogue>)`.

We can set it to `Optional[]` and it wouldn't be an issue right now since the graph/pattern designer is the only one that uses that interface directly and it's already optional at that stage.
Also, by *not* setting the storage it should default to a separate KeyValueStore and the only downside is that two agents in the same folder would overwrite the same default-created file. So for production level agents this should essentially not be optional. But at the same time, I think that this will be a topic that we need to get back to at a later stage.